Refactoring of the CLI interface (#291)
nodejs/corepack: README.md

Install the selected package managers and install them on the system.

Package managers thus installed will be configured as the new default when
calling their respective binaries outside of projects defining the
`packageManager` field.

### `corepack pack [--all] [... name@version]`

| Option                | Description                                |
| --------------------- | ------------------------------------------ |
| `--all`               | Pack all Last Known Good releases          |
| `--json `             | Print the output folder rather than logs   |
| `-o,--output `        | Path where to generate the archive         |

Download the selected package managers and store them inside a tarball
suitable for use with `corepack install -g`.

### `corepack use <name@version>`

When run, this command will retrieve the latest release matching the provided
descriptor, assign it to the project's package.json file, and automatically
perform an install.

### `corepack up`

Retrieve the latest available version for the current major release line of
the package manager used in the local project, and update the project to use
it.

> Unlike `corepack use` this command doesn't accept/need a version range, as it

Reviewer: ```suggestion
Unlike `corepack use` this command doesn't take a package manager name nor a version range, as it
```

will always select the latest available version from the same major line.
Should you need to upgrade to a new major, use an explicit

> `corepack use {name}@^{version}` call.

Reviewer: ```suggestion
`corepack use {name}@latest` call.
```

Reviewer: ```suggestion
`corepack use {name}@*` call.
```
